Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -376,6 +419,18 @@ void MediaPlayerPrivateGStreamerMSE::getSupportedTypes(HashSet<String, ASCIICase
 
 MediaPlayer::SupportsType MediaPlayerPrivateGStreamerMSE::supportsType(const MediaEngineSupportParameters& parameters)
 {
+    static unsigned mediaMaxWidth = 0;
+    static unsigned mediaMaxHeight = 0;
+    static unsigned mediaMaxFrameRate = 0;
+#ifdef VIDEO_DECODING_LIMIT
+    static bool isVideoDecoderLimitDefaultParsed = false;
```

Comment:
To avoid this, please use `std::once` and `std::once_flag`. Now that I see this code I think you can forget about using `std::optional<std::tuple<...>>`. I think the proper thing here would be capturing `&mediaMaxWidth, &mediaMaxHeight, &mediaMaxFrameRage` in the lamba that is going to be passed to the `std::once` call and inside that there you do the parsing.

I guess you also need to set a default somewhere and I think the proper place would be to do it in the declaration of the static variables.